test/e2e/certificates_test.go

```diff
@@ -0,0 +1,112 @@
+	// init test context
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	t.Run("Kubelet CA rotation", tc.testKubeletCARotation)
+}
+
+// testKubeletCARotation tests the rotation of the kubelet CA certificate by forcing the generation of a new CA
+// certificate and ensuring the content of the CA bundle in all Windows nodes contains the new certificate. Once the
+// rotation is triggered, WMCO detects the new CA certificate, merge it with the initial bundle, and copy the resulting
+// bundle to all Windows nodes.
+//
+// This test accounts for the kubelet CA automatic rotation that takes place at 80% of the initial certificate's
+// lifecycle (1 year).  Note that the old certificate is still present in the cluster until day 365, so clients may be
+// able to use it.
+func (tc *testContext) testKubeletCARotation(t *testing.T) {
+	// force CA rotation
+	err := tc.forceKubeApiServerCertificateRotation()
+	require.NoError(t, err)
+	// loop all Windows nodes
+	for _, winNode := range gc.allNodes() {
+		// spin-off subtests to ensure the CA bundle file is verified in all Windows nodes
+		t.Run("node/"+winNode.Name, func(t *testing.T) {
+			err := tc.waitForKubeletCACertificateInNode(&winNode)
+			assert.NoErrorf(t, err, "kubelet CA certificate should be present in node %S", winNode.Name)
+		})
+	}
+}
+
+// forceKubeApiServerCertificateRotation forces the certificate rotation by setting the `certificate-not-after`
+// annotation to nil in the `kube-apiserver-to-kubelet-signer` secret
+func (tc *testContext) forceKubeApiServerCertificateRotation() error {
+	// note that the slash had to be encoded as ~1 because it's a reference: https://www.rfc-editor.org/rfc/rfc6901#section-3
+	patchData := fmt.Sprint(`[{"op":"replace","path":"/metadata/annotations/auth.openshift.io~1certificate-not-after","value": null }]`)
+	// invoke request
+	_, err := tc.client.K8s.CoreV1().Secrets("openshift-kube-apiserver-operator").Patch(context.TODO(),
+		"kube-apiserver-to-kubelet-signer", types.JSONPatchType, []byte(patchData), meta.PatchOptions{})
+	return err
+}
+
+// waitForKubeletCACertificateInNode waits for the kubelet CA certificate to be present in the CA bundle file for the
+// given Windows node. After retry.Interval runs an SSH job to fetch the CA bundle, if the job fails or
+// kubelet CA is not present in the node, retries every retry.Interval until retry.Timeout is reached.
+func (tc *testContext) waitForKubeletCACertificateInNode(node *core.Node) error {
+	addr, err := controllers.GetAddress(node.Status.Addresses)
+	if err != nil {
+		return err
+	}
+	// CA bundle location in Windows node. i.e. "C:\k\kubelet-ca.crt"
+	caBundlePath := windows.GetK8sDir() + nodeconfig.KubeletClientCAFilename
+	// PowerShell command to fetch content in the file
+	command := fmt.Sprintf("Get-Content -Raw -Path %s", caBundlePath)
+	// wait retry.Interval and verify the CA bundle content, try if needed
+	return wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
```

Comment:
wait.PollImmediate should be used here if possible

Reply:
Not sure about that, in this case. I'm counting on the initial `retry.Interval` wait, for the kube-apiserver-operator to update the ConfigMap with the new certificate and WMCO to copy the new certificate over the Windows node.

Reply:
If it fails on the first try, it will just retry.
but in the case where it succeeds on the first try it will save our CI an amount of time == retry.Interval * nodecount

Reply:
There is a high probability to fail the first attempt because the file is not up to date, that is number one. 

Second, the first SSH call to fetch the content of `C:\k\kubelet-ca.crt` may block WMCO from performing the I/O operation required to copy the new certificate, and/or the kubelet service running in the node from dynamically loading the file.

To optimize the amount of time CI is idle, I feel we can reduce the retry time (currently `retry.Interval` => 15s)  in half, down to 7s. WDYT?